### PR TITLE
Upgrade to moon v17, and yarn v4.

### PR DESCRIPTION
**Pull request by milesj**
### Summary of Changes

Title is self-explanatory.

Just helping out!

### Edge Cases

List any common edge cases that you have considered and tested.

### Testing Steps

CI should handle most of it yeah?

### Checklist

Please make sure to review and check all of the following:

- [x] I've tested the changes and all tests pass.
- [ ] (if web) I've tested the changes on various desktop screen sizes to ensure responsiveness.
- [ ] (if mobile) I've tested the changes on both light and dark modes.

### CODE REVIEW

File: .yarnrc.yml [hunk 1]
```diff
@@ -1,7 +1,11 @@
-nodeLinker: node-modules
-
 checksumBehavior: update
 
-nmHoistingLimits: "workspaces"
+compressionLevel: mixed
+
+enableGlobalCache: false
+
```
Comment (milesj) [inline .yarnrc.yml:6]: these were all added by yarn

**Comment by milesj** [comment]
@Robinnnnn When you're freeeee

**Comment by Robinnnnn** [comment]
broooo you're a **freaking legend** @milesj 

will probably get around to this tomorrow; will likely merge a separate PR that will trigger merge conflicts

**Comment by Robinnnnn** [comment]
update: mind addressing the conflicts? 😅

**Comment by Robinnnnn** [comment]
@milesj copied your branch here and fixed merge conflicts: https://github.com/gallery-so/gallery/pull/2124

unfortunately, the mobile app stopped building so it's not in a mergeable state. unsure if it's related to moon upgrade or yarn upgrade.

**Pull request closed without merging**